language/move-stdlib/src/natives/debug.rs

```diff
@@ -35,9 +35,8 @@ fn native_print(
     debug_assert!(ty_args.len() == 1);
     debug_assert!(args.len() == 1);
 
-    // No-op if the feature flag is not present.
-    #[cfg(feature = "testing")]
-    {
+    // No-op if the feature flag is not present. or if in Rust release build
```

Comment:
If folks get the `move` or `sui` binaries from `cargo install` (recommended/most common install path), they will get Rust release binaries. If they then use these binaries in their local development/debugging flows, won't they get confused when `debug` printing doesn't work?

Reply:
> If folks get the `move` or `sui` binaries from `cargo install` (recommended/most common install path), they will get Rust release binaries. If they then use these binaries in their local development/debugging flows, won't they get confused when `debug` printing is a no-op?

We could conceivably include `--debug` flag by default with `cargo install` instructions and make a note that if someone wants faster local builds they can omit it but then they will loose the ability to debug-print outside of test code.

Reply:
Said differently: this means debug printing will only work if you build `move`/`sui` yourself from source, in debug mode. But this is not the flow we recommend/expect.

Reply:
> We could conceivably include --debug flag by default with cargo install instructions and make a note that if someone wants faster local builds they can omit it but then they will loose the ability to debug-print outside of test code.

I don't asking folks to re-install the `sui`/`move` binaries differently depending on whether they want to use debug printing/not (and asking them to use a slower binary when they want debug printing) is a good experience. Ideally, they should need to install one binary per version and it should just work for everything.

Reply:
Do you think making another feature flag would work?
We could make the CLI binaries have this flag on? (not sure if there is a way to do this).
I just want to make sure we don't accidentally turn this on for node software

Reply:
What is "this"--having `debug::print` work as a no-op?

Reply:
Running `debug::print` as an op.

We have cases where we want it to print
- unit testing
- some sort of local deployment
- maybe some sort of devnet

We need to be able to configure it to print in those cases. And we want to make sure it does not accidentally start spewing in some "release" build for a real testnet or mainnet.

Reply:
> As discussed this morning, this might be a reasonable way of letting debug prints occur on local deployments even when not in "test" mode

I think this problem was a nice-to-have for the future if we can figure out how to do it, but has to be connected to logging in the host system somehow (and I think a feature flag would be a good way). But I would vote on punting this problem for now and focusing on the Sui-side `debug` publishing issue